doc(perf-issues): Add uncompressed asset docs (#6153)
getsentry/sentry-docs: src/docs/product/issues/issue-details/performance-issues/uncompressed-asset/index.mdx

---
title: "Uncompressed Asset"
sidebar_order: 30
redirect_from:
  - /product/issues/performance-issues/uncompressed-asset/
description: "Learn more about Uncompressed Asset issues and how to diagnose and fix them."
---

<Include name="early-adopter-note.mdx" />

> Uncompressed asset issues are when one of the files your users are downloading to load your browser page was not compressed when transferred. This likely indicates a misconfiguration of the server or CDN (content delivery network) serving the asset file.
> 

Reviewer: ```suggestion
Uncompressed asset issues happen when a file that needs to be downloaded in order to load a browser page, doesn’t get compressed while being transferred. This may indicate a misconfiguration of the server or CDN (content delivery network) that’s serving the asset file.

```

## Detection Criteria

> The detector for this performance issue looks for specific asset (resource) spans with the following criteria:
> 
> - The span has to be one of either `resource.link` or `resource.script` spans, which in practice means it is a `.css` or `.js` file.

Reviewer: ```suggestion
The detector for this performance issue looks for specific asset (resource) spans with the following criteria:

- The span has to be one of either: `resource.link` or `resource.script` spans, which in practice means it's a `.css` or `.js` file.
```

> - The span has to be one of either `resource.link` or `resource.script` spans, which in practice means it is a `.css` or `.js` file.

Author: From what I've seen it's almost always, but not necessarily always. I added this in because people don't really understand 'span ops' but if you say `css / js` I think it's clearer what it's detecting.

> - It has to be over `512kb` in size, to rule out smaller files less likely to benefit from being compressed.
> - The span duration has to be over `500ms`, to rule out faster transfers or assets served in internal networks.
> - There has to be another asset span in the transaction that has compressed files. This helps rule out cases whether your users browser is not sending the `Accept-Encoding` header whatsoever.
> 
> If Sentry isn't detecting an uncompressed asset issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.

Reviewer: ```suggestion
- It has to be over `512kb` in size, (to filter out smaller files that are less likely to benefit from being compressed).
- The span duration has to be over `500ms`, (to filter out faster transfers or assets served in internal networks).
- There has to be another asset span in the transaction that has compressed files. This helps filter out cases where a users' browser isn't sending the `Accept-Encoding` header.

If Sentry isn't detecting an uncompressed asset issue where you expect one, it's probably because the transaction didn't meet one of the above criteria.
```

## Span Evidence

You can identify an Uncompressed Asset problem by four main aspects in the "Span Evidence" section:

- Transaction name
- Slow Resource Span - The uncompressed slow span, usually containing the asset url
- Asset Size - The size of the asset

> - Duration Impact - The fraction of time added to the total transaction time, which can likely be reduced with compression.

Reviewer 2: We discussed this in slack, might want to mention "maximum"

Author: 👍

> - Transaction name
> - Slow Resource Span - The uncompressed slow span, usually containing the asset url
> - Asset Size - The size of the asset
> - Duration Impact - The fraction of time added to the total transaction time, which can likely be reduced with compression.

Reviewer: ```suggestion
- Transaction name
- Slow resource span - The uncompressed slow span, usually containing the asset url
- Asset size 
- Duration impact - The fraction of time added to the total transaction time, which can likely be reduced with compression
```

> ![Uncompressed asset span evidence](span-evidence.png)

Reviewer 2: In the other docs, we omitted the span tree for span evidence, and we used the span tree in the example section. I think it makes sense to keep it consistent this way.

View it by going to the **Issues** page in Sentry, selecting your project, clicking on the Uncompressed Asset error you want to examine, then scrolling down to the "Span Evidence" section in the "Details" tab.

## Recommendations

> Check the server or CDN (content delivery network) where the asset is being hosted and make sure it is setup for gzip or brotli compression.

Reviewer: ```suggestion
Check the server or CDN (content delivery network) where the asset is being hosted and make sure it's set up for gzip or brotli compression.
```
